woocommerce/woocommerce
# Add generic function to determine if URL is a store page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/45110.

The function `is_store_page` is added to `WCAdminHelper` and is used to determine if a URL is a store page.

Store pages are defined as:

- Shop Page
- Cart Page
- Checkout Page
- My Account Page
- Terms and Conditions Page
- Privacy Policy Page

Additionally, the following autogenerated pages should be included:

- Product
- Product Categories (taxonomy=product_cat)
- Product Tags (taxonomy=product_tag)

The function ignores the domain and protocol of the URL and only checks the path and query string so that users can only pass `$_SERVER['REQUEST_URI']` to the function for checking if the current page is a store page.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site
2. Install `WooCommerce Smooth Generator` and `Code Snippets` plugins
3. Either skip or complete Core profiler
4. Go to `/wp-admin/tools.php?page=smoothgenerator` > Generate 10 products  OR run `wp wc generate products 10`
5. Go to `/wp-admin/admin.php?page=snippets`
6. Add new Snippet

```php
use Automattic\WooCommerce\Admin\WCAdminHelper;

function maybe_add_banner_to_footer() {
	if ( wp_doing_ajax() || wp_doing_cron() || WC()->is_rest_api_request() ) {
		return;
	}

	$is_store = WCAdminHelper::is_store_page();
	if ( $is_store ) {
    	echo '<div style="position: fixed; bottom: 0; background-color: red; color: white; width: 100%; text-align: center;">This is a store page!</div>';
	}
}

add_action('wp_footer', 'maybe_add_banner_to_footer');
```

7.  Select `Only run on site front-end` and activate
8. Go to frontend
9. Confirm the `This is a store page!` red banner is displayed on the store pages (Shop, Cart, Checkout, My Account, Terms and Conditions, Privacy Policy, Product, Product Categories, Product Tags) and not displayed on other pages (Home, Sample Page, etc.)
10. Go to `/wp-admin/admin.php?page=wc-settings&tab=advanced`
11. Change `Terms and conditions` page to Sample Page
12. Go to Sample Page
13. Confirm the banner is displayed on the Sample Page
14. Go to `/wp-admin/options-permalink.php`
15. Change Product permalinks to `Shop base with category`
16. Change Product category base to `test-category`
17. Go to a product page
18. Confirm the banner is still displayed on the product page
19. Go to a product category page

<img width="750" alt="Screenshot 2024-03-05 at 11 32 37" src="https://github.com/woocommerce/woocommerce/assets/4344253/f3d7c5c5-4faf-4927-baa5-ef505e54f488">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
